app/services/safe_downloader.rb

```diff
@@ -13,7 +13,8 @@ class SafeDownloader
     OpenURI::HTTPError,
     RuntimeError,
     URI::InvalidURIError,
-    DownloadError
+    DownloadError,
+    TypeError
```

Comment:
This is raised when `IO.read` tries to read something it can't:

```

E, [2019-06-11T13:03:43.412502 #1] ERROR -- : [[compliance-reports-consumer] {platform.upload.compliance: 2, 1, 0}:] Exception raised when processing platform.upload.compliance/1 at offset 5451 -- TypeError: no implicit conversion of StringIO into String
--
  | /opt/app-root/src/app/services/safe_downloader.rb:22:in `read'
  | /opt/app-root/src/app/services/safe_downloader.rb:22:in `download'
```